Sources/S3Storage/S3Adapter.swift

```diff
@@ -59,26 +59,42 @@ public class S3Adapter: Adapter {
     /// Used within the AWS HMAC signing
     let service = "s3"
     
+    let host: URL
+    
     /// Create a new Local adapter.
-    public init(accessKey: String, secretKey: String, region: Region, securityToken: String? = nil) throws {
+    public init(host: URL = URL(string: "https://s3.amazonaws.com")!, accessKey: String, secretKey: String, region: Region, securityToken: String? = nil) throws {
```

Comment:
I'd change this from `URL` to `URLRepresentable` 
`public init(host: URLRepresentable = "https://s3.amazonaws.com", accessKey: String, secretKey: String, region: Region, securityToken: String? = nil) throws`